### DISCUSSION TITLE
Re-implementing OpenPGP with CryptoSwift?

### DISCUSSION THREAD

**Discussion by ikester** [krzyzanowskim/CryptoSwift#996] [Q&A]
I've been looking for a pure Swift implementation of OpenPGP functionality. I'm aware of Marcin's ObjectivePGP project but I was wondering if anyone is working on a Swift OpenPGP library, perhaps using CryptoSwift?

Has anyone considered adding OpenPGP functionality to CryptoSwift?

**Comment by krzyzanowskim** [comment]
Is there a problem with ObjectivePGP use with Swift? PGP is nasty and has a lot of old and new requirements. That all makes it hard (or more precisely, very niche) to reimplement just in Swift. I would argue whether it's the best investment of spare time ;) but also I already did it so I may be biased.

**Comment by ikester** [comment]
Thanks for all your crypto work @krzyzanowskim. ObjectivePGP should work fine in Swift but it has a less permissible license than CryptoSwift.

I'm not looking for 100% coverage of all the OpenPGP spec but, at the very minimum, the ability to manage PGP keys, encrypt and decrypt messages, and verify signatures. PGP keys and signatures are pretty ubiquitous and it would be great to have a Swift way of dealing with them.

If you could do it all over again, how would you do public key crypto and signatures with modern crypto primitives?
